src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,64 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+// TODO the commonalities does not seem to do have anything to do with contact information but rather
+// some kind of validation, perhaps an interface something like Validated might be more suitable.
+// Alternatively, Name actually can also be considered a Contact.
```

Comment:
Yea but I was thinking if I want to have multiple interfaces or a parent class, because I can't if I implemented a Validated parent class to ensure that validation is implemented for subclasses.